stdlib/ballerina-http/src/main/java/org/ballerinalang/net/uri/URIUtil.java

```diff
@@ -62,4 +68,52 @@ public static void populateQueryParamMap(String queryParamString, BMap<String, B
             }
         }
     }
+
+    public static BMap<String, BValue> getMatrixParamsMap(String path, HTTPCarbonMessage carbonMessage) {
+        BMap<String, BValue> matrixParamsBMap = new BMap<>();
+        Map<String, Map<String, String>> pathToMatrixParamMap =
+                (Map<String, Map<String, String>>) carbonMessage.getProperty(Constants.MATRIX_PARAMS);
+        if (!pathToMatrixParamMap.containsKey(path)) {
+            return matrixParamsBMap;
+        }
+        Map<String, String> matrixParamsMap = pathToMatrixParamMap.get(path);
+        for (Map.Entry<String, String> matrixParamEntry : matrixParamsMap.entrySet()) {
+            matrixParamsBMap.put(matrixParamEntry.getKey(), new BString(matrixParamEntry.getValue()));
+        }
+        return matrixParamsBMap;
+    }
+
+
+    public static String removeMatrixParams(String path, Map<String, Map<String, String>> matrixParams) {
+        if (path.startsWith("/")) {
+            path = path.substring(1);
+        }
+        String[] pathSplits = path.split("\\?");
+        String[] pathSegments = pathSplits[0].split("/");
+        String pathToMatrixParam = "";
+        for (String pathSegment : pathSegments) {
+            String[] splitPathSegment = pathSegment.split(";");
+            pathToMatrixParam = pathToMatrixParam.concat("/" + splitPathSegment[0]);
+            Map<String, String> segmentMatrixParams = new HashMap<>();
+            if (splitPathSegment.length > 1) {
+                for (int i = 1; i < splitPathSegment.length; i++) {
+                    String[] splitMatrixParam = splitPathSegment[i].split("=");
+                    if (splitMatrixParam.length != 2) {
+                        throw new BallerinaConnectorException(
+                                String.format("Found non-matrix parameter '%s' in path '%s'",
+                                              splitPathSegment[i], path));
+                    }
+                    segmentMatrixParams.put(splitMatrixParam[0], splitMatrixParam[1]);
+                }
+            }
+            matrixParams.put(pathToMatrixParam, segmentMatrixParams);
+        }
+
+        if (pathSplits.length > 1) {
```

Comment:
Do we need this if check here?

Reply:
Since query parameters are optional there is a high chance where pathSplit.length = 1

Reply:
Got your point. Will make the changes. Thanks :)